src/esp/nav/test/PathFinderTest.cpp

```diff
@@ -72,6 +78,67 @@ void PathFinderTest::tryStepNoSliding() {
   }
 }
 
+void PathFinderTest::multiGoalPath() {
+  esp::nav::PathFinder pathFinder;
+  pathFinder.loadNavMesh(skokloster);
+  CORRADE_VERIFY(pathFinder.isLoaded());
+  pathFinder.seed(0);
+
+  for (int __j = 0; __j < 1000; ++__j) {
+    std::vector<esp::vec3f> points;
+    for (int i = 0; i < 10; ++i) {
+      points.emplace_back(pathFinder.getRandomNavigablePoint());
+    }
+
+    esp::nav::MultiGoalShortestPath multiPath;
+    multiPath.requestedStart = points[0];
+    multiPath.requestedEnds = {points.begin() + 1, points.end()};
+
+    CORRADE_VERIFY(pathFinder.findPath(multiPath));
+
+    esp::nav::MultiGoalShortestPath path;
+    path.requestedStart = points[0];
+    float trueMinDist = 1e5;
+    for (int i = 1; i < points.size(); ++i) {
+      path.requestedEnds = {points[i]};
+
+      CORRADE_VERIFY(pathFinder.findPath(path));
+
+      trueMinDist = std::min(trueMinDist, path.geodesicDistance);
+    }
+
+    CORRADE_COMPARE(multiPath.geodesicDistance, trueMinDist);
+  }
+}
+
+void PathFinderTest::pathFindBenchmark() {
+  esp::nav::PathFinder pathFinder;
+  pathFinder.loadNavMesh(skokloster);
+  CORRADE_VERIFY(pathFinder.isLoaded());
+
+  if (testCaseInstanceId() == 0) {
+    setTestCaseDescription("Path to single point");
+    esp::nav::ShortestPath path;
+    path.requestedStart = pathFinder.getRandomNavigablePoint();
```

Comment:
Does that make sense to calculate distance to the same 100 points using single shortest path search?